Cannot install on Fedora 31 with kernel 5.5.7-200

Error occurs when installing `fedora-31-displaylink-1.6.4-2.x86_64.rpm`. [`/var/log/displaylink/displaylink.log`](https://github.com/displaylink-rpm/displaylink-rpm/files/4303056/displaylink.log) and 
[`/var/lib/dkms/evdi/1.6.4/build/make.log`](https://github.com/displaylink-rpm/displaylink-rpm/files/4303057/make.log) attached.

thanks for that.  I followed the steps & got this:

```
[root@Revan:displaylink-rpm]# make rawhide
Checking last upstream commit date...
make RELEASE="`        echo -e "1" | tr -d '[:space:]'`.rawhide.`	curl -s https://api.github.com/repos/DisplayLink/evdi/branches/devel -H "Accept: application/vnd.github.full+json" | grep date | head -1 | cut -d: -f 2- | sed s/[^0-9TZ]//g`" devel all
make[1]: Entering directory '/root/displaylink-rpm'
git clone --depth 1 -b devel https://github.com/DisplayLink/evdi.git /var/tmp/evdi-1.6.4
Cloning into '/var/tmp/evdi-1.6.4'...
remote: Enumerating objects: 45, done.
remote: Counting objects: 100% (45/45), done.
remote: Compressing objects: 100% (45/45), done.
remote: Total 45 (delta 1), reused 8 (delta 0), pack-reused 0
Unpacking objects: 100% (45/45), done.
cd /var/tmp/evdi-1.6.4 && git pull
Already up to date.
tar -z -c -f v1.6.4.tar.gz -C /var/tmp evdi-1.6.4
rpmbuild -bb --define "_topdir `pwd`" --define "_sourcedir `pwd`" --define "_rpmdir `pwd`" --define "_specdir `pwd`" --define "_srcrpmdir `pwd`" --define "_buildrootdir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_builddir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_release 1.rawhide.20200303T134631Z" --define "_daemon_version 5.2.14" --define "_version 1.6.4" --define "_tmppath `mktemp -d /var/tmp/displayportXXXXXX`"  displaylink.spec --target=i386
/bin/sh: rpmbuild: command not found
make[1]: *** [Makefile:132: i386/displaylink-1.6.4-1.rawhide.20200303T134631Z.i386.rpm] Error 127
make[1]: Leaving directory '/root/displaylink-rpm'
make: *** [Makefile:78: rawhide] Error 2
```

Got an aerror.  tried to start from scratch & re-clone the git.  Still errored out on me:

```[root@Revan:displaylink-rpm]# make rawhide
Checking last upstream commit date...
make RELEASE="`        echo -e "1" | tr -d '[:space:]'`.rawhide.`	curl -s https://api.github.com/repos/DisplayLink/evdi/branches/devel -H "Accept: application/vnd.github.full+json" | grep date | head -1 | cut -d: -f 2- | sed s/[^0-9TZ]//g`" devel all
make[1]: Entering directory '/root/displaylink-rpm'
cd /var/tmp/evdi-1.6.4 && git pull
Already up to date.
tar -z -c -f v1.6.4.tar.gz -C /var/tmp evdi-1.6.4
wget --post-data="fileId=1369    &accept_submit=Accept" -O DisplayLink\ USB\ Graphics\ Software\ for\ Ubuntu\ 5.2.14.zip \
	 https://www.displaylink.com/downloads/file?id=1369    
--2020-03-09 13:10:44--  https://www.displaylink.com/downloads/file?id=1369
Resolving www.displaylink.com (www.displaylink.com)... 34.252.102.94, 52.209.94.109, 52.214.95.115, ...
Connecting to www.displaylink.com (www.displaylink.com)|34.252.102.94|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: http://assets.displaylink.com/live/downloads/software/f1369_DisplayLink%20USB%20Graphics%20Software%20for%20Ubuntu%205.2.zip?AWSAccessKeyId=AKIAJHGQWPVXWHEDJUEA&Expires=1583778045&Signature=i%2FarirBVkdjQQJw%2F4pU9yCXnqdM%3D [following]
--2020-03-09 13:10:45--  http://assets.displaylink.com/live/downloads/software/f1369_DisplayLink%20USB%20Graphics%20Software%20for%20Ubuntu%205.2.zip?AWSAccessKeyId=AKIAJHGQWPVXWHEDJUEA&Expires=1583778045&Signature=i%2FarirBVkdjQQJw%2F4pU9yCXnqdM%3D
Resolving assets.displaylink.com (assets.displaylink.com)... 52.218.100.123
Connecting to assets.displaylink.com (assets.displaylink.com)|52.218.100.123|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 12768730 (12M) [application/octet-stream]
Saving to: ‘DisplayLink USB Graphics Software for Ubuntu 5.2.14.zip’

DisplayLink USB Graphics Software 100%[==========================================================>]  12.18M   463KB/s    in 25s     

2020-03-09 13:11:11 (494 KB/s) - ‘DisplayLink USB Graphics Software for Ubuntu 5.2.14.zip’ saved [12768730/12768730]

rpmbuild -bb --define "_topdir `pwd`" --define "_sourcedir `pwd`" --define "_rpmdir `pwd`" --define "_specdir `pwd`" --define "_srcrpmdir `pwd`" --define "_buildrootdir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_builddir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_release 1.rawhide.20200303T134631Z" --define "_daemon_version 5.2.14" --define "_version 1.6.4" --define "_tmppath `mktemp -d /var/tmp/displayportXXXXXX`"  displaylink.spec --target=i386
/bin/sh: rpmbuild: command not found
make[1]: *** [Makefile:132: i386/displaylink-1.6.4-1.rawhide.20200303T134631Z.i386.rpm] Error 127
make[1]: Leaving directory '/root/displaylink-rpm'
make: *** [Makefile:78: rawhide] Error 2
```

i have a similar problem on a later kernel:
`$ modprobe evdi
modprobe: FATAL: Module evdi not found in directory /lib/modules/5.5.15-200.fc31.x86_64`
`$ rpm -qa | grep kernel-headers
kernel-headers-5.5.15-200.fc31.x86_64`

and when i follow the instructions from #119  i get this error:

```
.........................
Saving to: ‘DisplayLink USB Graphics Software for Ubuntu 5.2.14.zip’
2020-04-10 14:34:58 (427 KB/s) - ‘DisplayLink USB Graphics Software for Ubuntu 5.2.14.zip’ saved [12768730/12768730]

rpmbuild -bb --define "_topdir `pwd`" --define "_sourcedir `pwd`" --define "_rpmdir `pwd`" --define "_specdir `pwd`" --define "_srcrpmdir `pwd`" --define "_buildrootdir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_builddir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_release 1.rawhide.20200303T134631Z" --define "_daemon_version 5.2.14" --define "_version 1.6.4" --define "_tmppath `mktemp -d /var/tmp/displayportXXXXXX`"  displaylink.spec --target=i386
/bin/sh: rpmbuild: command not found
make[1]: *** [Makefile:132: i386/displaylink-1.6.4-1.rawhide.20200303T134631Z.i386.rpm] Error 127
make[1]: Leaving directory '/home/behshad/Downloads/displaylink-rpm'
make: *** [Makefile:78: rawhide] Error 2
```

and it stops there, any suggestions?

> i have a similar problem on a later kernel:
> `$ modprobe evdi modprobe: FATAL: Module evdi not found in directory /lib/modules/5.5.15-200.fc31.x86_64`
> `$ rpm -qa | grep kernel-headers kernel-headers-5.5.15-200.fc31.x86_64`
> 
> and when i follow the instructions from #119 i get this error:
> 
> ```
> .........................
> Saving to: ‘DisplayLink USB Graphics Software for Ubuntu 5.2.14.zip’
> 2020-04-10 14:34:58 (427 KB/s) - ‘DisplayLink USB Graphics Software for Ubuntu 5.2.14.zip’ saved [12768730/12768730]
> 
> rpmbuild -bb --define "_topdir `pwd`" --define "_sourcedir `pwd`" --define "_rpmdir `pwd`" --define "_specdir `pwd`" --define "_srcrpmdir `pwd`" --define "_buildrootdir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_builddir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_release 1.rawhide.20200303T134631Z" --define "_daemon_version 5.2.14" --define "_version 1.6.4" --define "_tmppath `mktemp -d /var/tmp/displayportXXXXXX`"  displaylink.spec --target=i386
> /bin/sh: rpmbuild: command not found
> make[1]: *** [Makefile:132: i386/displaylink-1.6.4-1.rawhide.20200303T134631Z.i386.rpm] Error 127
> make[1]: Leaving directory '/home/behshad/Downloads/displaylink-rpm'
> make: *** [Makefile:78: rawhide] Error 2
> ```
> 
> and it stops there, any suggestions?

actually problem above was due to some dependencies which i installed and successfully did the #119 , however second USB screen is still not working:

```
$ systemctl status displaylink.service
displaylink.service - DisplayLink Manager Service
   Loaded: loaded (/usr/lib/systemd/system/displaylink.service; static; vendor preset: disabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2020-04-10 14:53:26 NZST; 1s ago
  Process: 4594 ExecStartPre=/sbin/modprobe evdi (code=exited, status=1/FAILURE)
      CPU: 7ms
```

```
$ systemctl status dlm.service
dlm.service - dlm control daemon
   Loaded: loaded (/usr/lib/systemd/system/dlm.service; enabled; vendor preset: disabled)
   Active: inactive (dead)

Apr 10 14:51:45 localhost.localdomain systemd[1]: Dependency failed for dlm control daemon.
Apr 10 14:51:45 localhost.localdomain systemd[1]: dlm.service: Job dlm.service/start failed with result 'de>
```
and finally
```
$ xrandr --listproviders
Providers: number : 1
Provider 0: id: 0x44 cap: 0xf, Source Output, Sink Output, Source Offload, Sink Offload crtcs: 3 outputs: 2 associated providers: 0 name:modesetting
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,15 +2,15 @@ language: generic
 sudo: required
 env:
   global:
-    - VERSION=1.6.4
-    - DAEMON_VERSION=5.2.14
-    - RELEASE=3
+    - VERSION=1.7.0
+    - DAEMON_VERSION=5.3.1.34
+    - RELEASE=1
   matrix:
+  - OS_TYPE=fedora OS_VERSION=32
   - OS_TYPE=fedora OS_VERSION=31
   - OS_TYPE=fedora OS_VERSION=30
   - OS_TYPE=fedora OS_VERSION=29
   - OS_TYPE=fedora OS_VERSION=28
-  - OS_TYPE=fedora OS_VERSION=27
   - OS_TYPE=centos OS_VERSION=7
 services:
 - docker

--- a/Makefile
+++ b/Makefile
@@ -2,10 +2,10 @@
 # Versions
 #
 
-DAEMON_VERSION := 5.2.14
-DOWNLOAD_ID    := 1369    # This id number comes off the link on the displaylink website
-VERSION        := 1.6.4
-RELEASE        := 3
+DAEMON_VERSION := 5.3.1.34
+DOWNLOAD_ID    := 1576    # This id number comes off the link on the displaylink website
+VERSION        := 1.7.0
+RELEASE        := 1
 
 #
 # Dependencies

--- a/displaylink.spec
+++ b/displaylink.spec
@@ -24,7 +24,7 @@ ExclusiveArch:	i386 x86_64
 BuildRequires:  gcc-c++
 BuildRequires:	libdrm-devel
 BuildRequires:  make
-Requires:       dkms, %{kernel_pkg_name} > 4.7, %{kernel_pkg_name}-devel > 4.7
+Requires:       dkms, %{kernel_pkg_name} >= 4.15, %{kernel_pkg_name}-devel >= 4.15
 Conflicts:      xorg-x11-server-Xorg = 1.20.1
 
 %description
@@ -84,7 +84,7 @@ cp -a x86-ubuntu-1604/DisplayLinkManager $RPM_BUILD_ROOT/usr/libexec/displaylink
 %endif
 
 # Firmwares
-cp -a ella-dock-release.spkg firefly-monitor-release.spkg $RPM_BUILD_ROOT/usr/libexec/displaylink/
+cp -a ella-dock-release.spkg firefly-monitor-release.spkg ridge-dock-release.spkg $RPM_BUILD_ROOT/usr/libexec/displaylink/
 
 # systemd/udev
 cp -a %{SOURCE1} $RPM_BUILD_ROOT/usr/lib/systemd/system/
@@ -101,13 +101,11 @@ chmod +x $RPM_BUILD_ROOT/usr/lib/systemd/system-sleep/displaylink.sh
 /usr/bin/systemctl -q is-active displaylink.service && /usr/bin/systemctl stop displaylink.service
 /usr/bin/systemctl daemon-reload
 /usr/bin/systemctl -q is-enabled dkms.service || /usr/bin/systemctl enable dkms.service
+/sbin/dkms add evdi/%{version} --rpm_safe_upgrade >> %{logfile} 2>&1
+/sbin/dkms build evdi/%{version} >> %{logfile} 2>&1
 /sbin/dkms install evdi/%{version} >> %{logfile} 2>&1
 ln -s /usr/libexec/displaylink/libevdi.so.%{version} /usr/libexec/displaylink/libevdi.so
 /usr/bin/systemctl start displaylink.service
-
-%triggerin -- kernel
-NEWEST_KERNEL=$(rpm -q kernel|sort -V|head -1|cut -d- -f2-)
-/sbin/dkms install evdi/%{version} ${NEWEST_KERNEL} >>%{logfile} 2>&1
 
 %files
 %doc LICENSE
@@ -124,14 +122,22 @@ NEWEST_KERNEL=$(rpm -q kernel|sort -V|head -1|cut -d- -f2-)
 %preun
 if [ $1 -eq 0 ] ;then
 	/usr/bin/systemctl -q is-active displaylink.service && /usr/bin/systemctl stop displaylink.service
-        %{__rm} -f /usr/libexec/displaylink/libevdi.so /usr/libexec/displaylink/libevdi.so.%{version}
-	/sbin/dkms remove evdi/%{version} --all >> %{logfile}
+	%{__rm} -f /usr/libexec/displaylink/libevdi.so
+	/sbin/dkms remove evdi/%{version} --all --rpm_safe_upgrade >> %{logfile}
 fi
 
 %postun
 /usr/bin/systemctl daemon-reload
 
 %changelog
+* Mon May 04 2020 Michael L. Young <elgueromexicano@gmail.com> 1.7.0-1
+- Update to evdi driver version 1.7.0.
+- Update to Displaylink driver 5.3.1.
+- The minimum kernel supported in evdi is now 4.15. Adjusting spec to match.
+- Fix support for DL-6xxx devices. The firmware image was not being copied from
+  the DisplayLink driver package.
+- Adjust how we use dkms inside the rpm to follow recommended way in documentation.
+
 * Thu Apr 16 2020 Michael L. Young <elgueromexicano@gmail.com> 1.6.4-3
 - Remove patches that are no longer needed.  This restores the ability
   to build against rawhide.

